hmily-tac/hmily-tac-sqlcompute/src/main/java/org/dromara/hmily/tac/sqlcompute/impl/HmilyUpdateSQLComputeEngine.java

```diff
@@ -74,11 +74,17 @@ private String getWhereCondition(final String sql) {
         return sqlStatement.getWhere().map(segment -> sql.substring(segment.getStartIndex(), segment.getStopIndex() + 1)).orElse("");
     }
     
-    private Optional<Object> getExpressionParameter(final List<Object> parameters, final HmilyExpressionSegment expressionSegment) {
-        if (expressionSegment instanceof HmilyParameterMarkerExpressionSegment) {
-            return Optional.of(parameters.get(((HmilyParameterMarkerExpressionSegment) expressionSegment).getParameterMarkerIndex()));
-        }
-        return Optional.empty();
+    private List<Object> getWhereParameters(final List<Object> parameters) {
```

Comment:
seems there exist some problems in getWhereParameters implement

Reply:
Thanks. I will fix the potential in the next pull request.